weight_formats/quantisation_training.py

```diff
@@ -235,6 +235,7 @@ def convert(
     scaling_mode: ScalingMode,
     clip_gradient: bool,
     error_weight: dict[str, Tensor] | None,
+    perturb_only: bool = False,
```

Comment:
I think this is all grand - no concerns about the functionality. Perhaps my only (very minor) comment is the name, maybe could consider `mode: Literal["qat", "one-shot"] = "qat"`, or something like that?